### PR TITLE
Add Pure Storage Protection Groups module

### PR DESCRIPTION
**Pull request by sdodsley**
SUMMARY

Storage module for Pure Storage FlashArray device. Module to create/delete protection groups

ISSUE TYPE

New Module Pull Request
COMPONENT NAME

purefa_pg
ANSIBLE VERSION

ansible 2.4.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/pureuser/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]

### CODE REVIEW

**Comment by sdodsley** [comment]
@gundalow As requested separate PR for each remaining module

**Comment by sdodsley** [comment]
@gundalow  : Can you review, please?

File: lib/ansible/modules/storage/purestorage/purefa_pg.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+def main():
+    argument_spec = purefa_argument_spec()
+    argument_spec.update(
+        dict(
+            pgroup=dict(required=True),
+            state=dict(default='present', choices=['present', 'absent']),
+            volume=dict(type='list'),
+            host=dict(type='list'),
+            hostgroup=dict(type='list'),
+            eradicate=dict(default='false', choices=['false', 'true']),
```
Comment (gundalow) [inline lib/ansible/modules/storage/purestorage/purefa_pg.py]: `eradicate=dict(default='false', type='bool),`

File: lib/ansible/modules/storage/purestorage/purefa_pg.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  hostgroup:
+    description:
+      - List of existing hostgroups to add to protection group.
+    required: false
+  eradicate:
+    description:
+      - Define whether to eradicate the protection group on delete and leave in trash.
+    required: false
+    default: false
+    choices: [ "false", "true" ]
```
Comment (gundalow) [inline lib/ansible/modules/storage/purestorage/purefa_pg.py]: Please replace with `type: bool`

http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-block

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by sdodsley** [comment]
@gundalow  changes completed as requested

**Review by gundalow — APPROVED** [review]

**Comment by sdodsley** [comment]
@gundalow changes added as per comment. Please review together with PR #26355

**Comment by sdodsley** [comment]
:shipit:

**Comment by gundalow** [comment]
Merged, and will be released in Ansible 2.4.0

**Pull request merged**